[docs] Update python instructions in contributor guide (#17699)
yugabyte/yugabyte-db: docs/content/preview/contribute/docs/docs-build.md

> * **Python**: You need `python` to be available somewhere in your shell path.

Reviewer: ```suggestion
* **Python** (version 3.10 or earlier): You need `python` to be available somewhere in your shell path.
```

>     Recent versions of macOS have only a `python3` executable, as does the Homebrew install. You can use [pyenv](https://github.com/pyenv/pyenv) to manage multiple versions of python on your system. Make sure you point to a python version less than 3.10.

Reviewer: ```suggestion
    Recent versions of macOS have only a `python3` executable, as does the Homebrew install. You can use [pyenv](https://github.com/pyenv/pyenv) to manage multiple versions of python on your system. Make sure to point to Python version 3.10 or earlier.
```
